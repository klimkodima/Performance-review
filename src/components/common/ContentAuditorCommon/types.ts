export type ImageType = {
  classes: string;
  src: string;
  alt: string;
};
