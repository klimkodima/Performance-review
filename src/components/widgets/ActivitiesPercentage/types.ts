export interface IActivity {
  value: number;
  name: string;
}

export interface ICardActivity {
  activities: IActivity[];
  totalTime: number;
}

export interface IGridBox {
  xs: number;
  md: number;
  lg: number;
}

export interface IPieOption {
  title: {
    text: string;
    textStyle: {
      fontWeight: string;
      fontSize: string;
    };
    top: string;
  };
  animation: false;
  color: string[];
  labelLine: {
    color: string;
  };
  label: {
    show: true;
    color: string;
    fontWeight: 500;
    fontSize: string;
    formatter: string;
  };
  series: [
    {
      name: string;
      type: string;
      radius: string;
      data: IActivity[];
      labelLine: {
        lineStyle: {
          color: string;
        };
      };
      emphasis: {
        disabled: boolean;
      };
      top: string;
    }
  ];
}
