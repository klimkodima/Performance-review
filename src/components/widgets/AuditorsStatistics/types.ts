export enum ROLE {
  TeamLead = 'TEAM_LEAD',
  Auditor = 'AUDITOR',
  Manager = 'MANAGER'
}

export enum ORDER {
  Asc = 'ASC',
  Desc = 'DESC'
}

export interface ITableLabel {
  text: string;
  sorted?: boolean;
}

export interface INumberColValues {
  totalPoints: number;
  averagePerformance: number;
  averageValidity: number;
  supportAuditsHours: number;
  correspondentsHours: number;
}

export interface IAuditorStatistics extends INumberColValues {
  id: number;
  fullName: string;
}
