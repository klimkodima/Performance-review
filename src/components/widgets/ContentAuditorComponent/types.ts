export type RowType = {
  classesContainer: string;
  classesLabel: string;
  classesText: string;
  label: string;
  text: string;
};
