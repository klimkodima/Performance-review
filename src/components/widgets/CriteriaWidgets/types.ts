export type TData = {
  title: string;
  statisticValue: number;
  isCorrespondence: boolean;
};

export type TDataList = TData[];
