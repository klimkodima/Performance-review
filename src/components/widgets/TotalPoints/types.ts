export type TTotalPoints = {
  totalPoints: number;
  level: number;
  premium: number;
};
