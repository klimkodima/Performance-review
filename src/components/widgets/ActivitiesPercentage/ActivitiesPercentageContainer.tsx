import { ReactElement } from 'react';
import Grid from '@mui/material/Grid';
import { useTranslation } from 'react-i18next';

import ActivitiesPercentage from './ActivitiesPercentage';

import { IActivity, ICardActivity, IPieOption } from './types';

const audits: IActivity = { value: 99, name: 'Audits' };
const meetings: IActivity = { value: 63, name: 'Meetings' };
const others: IActivity = { value: 0, name: 'Others' };
const support: IActivity = { value: 18, name: 'Support' };
export const PIE_TITLE_TEXT = 'Activities Percentage';
const colors = [
  '#F8D84F',
  '#94DC2E',
  '#9F00BB',
  '#67E0E3',
  '#ff9f7f',
  '#fb7293',
  '#E062AE',
  '#E690D1',
  '#e7bcf3',
  '#9d96f5',
  '#8378EA',
  '#96BFFF'
];

export const activities: IActivity[] = [audits, others, support, meetings];

export const totalTime: number = activities.reduce(
  (acc: number, a: IActivity) => {
    return acc + a.value;
  },
  0
);

const ActivitiesPercentageContainer = (): ReactElement | null => {
  const { t } = useTranslation('common', { keyPrefix: 'activitiesPercentage' });
  const pieData = activities
    .filter((act) => act.value !== 0)
    .map((activity: IActivity) => ({
      ...activity,
      value: Math.round((activity.value / totalTime) * 100)
    }));

  const cardData: ICardActivity = {
    totalTime: totalTime,
    activities: activities
      .filter((act) => act.value !== 0)
      .sort((a: IActivity, b: IActivity) => b.value - a.value)
  };

  const option: IPieOption = {
    title: {
      text: t('Title'),
      textStyle: {
        fontWeight: '700',
        fontSize: '2rem'
      },
      top: '40px'
    },
    animation: false,
    color: colors,
    labelLine: {
      color: 'black'
    },
    label: {
      show: true,
      color: 'black',
      fontWeight: 500,
      fontSize: '1.3rem',
      formatter: '{b}: {d}%'
    },
    series: [
      {
        name: PIE_TITLE_TEXT,
        type: 'pie',
        radius: '57%',
        data: pieData,
        labelLine: {
          lineStyle: {
            color: 'black'
          }
        },
        emphasis: {
          disabled: true
        },
        top: '10%'
      }
    ]
  };

  return (
    <Grid item xs={12} md={12} lg={12}>
      <ActivitiesPercentage pieData={option} cardData={cardData} />
    </Grid>
  );
};

export default ActivitiesPercentageContainer;
