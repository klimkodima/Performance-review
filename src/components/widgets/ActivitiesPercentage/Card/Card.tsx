import { FC } from 'react';
import { useTranslation } from 'react-i18next';

import { ICardActivity } from '../types';

type CardProps = {
  cardData: ICardActivity;
};

const CardView: FC<CardProps> = ({ cardData }) => {
  const { t } = useTranslation('common', { keyPrefix: 'activitiesPercentage' });

  return (
    <div className='card'>
      <div className='card-row'>
        <label data-testid='totalTimeLabel' className='card-label'>
          {t('Total time')}:
        </label>
        <span data-testid='totalTime' className='card-value'>
          {String(cardData.totalTime)}h
        </span>
      </div>
      {cardData.activities.map((activity) => (
        <div key={activity.name} className='card-row'>
          <label className='card-label'>{activity.name}:</label>
          <span className='card-value'>{activity.value}h</span>
        </div>
      ))}
    </div>
  );
};

export default CardView;
