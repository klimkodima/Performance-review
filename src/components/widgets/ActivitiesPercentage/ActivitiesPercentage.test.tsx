import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { toMatchImageSnapshot } from 'jest-image-snapshot';
import { render, screen } from '@testing-library/react';
import puppeteer from 'puppeteer';

import { Card } from './Card';
import { activities, totalTime } from './ActivitiesPercentageContainer';

import { IActivity, ICardActivity } from './types';

const zeroAudits: IActivity = { value: 0, name: 'Audits' };
const zeroMeetings: IActivity = { value: 0, name: 'Meetings' };
const zeroOthers: IActivity = { value: 0, name: 'Others' };
const zeroSupport: IActivity = { value: 0, name: 'Support' };
const zeroAct = [zeroAudits, zeroMeetings, zeroOthers, zeroSupport];

const zeroCardData: ICardActivity = {
  activities: zeroAct
    .filter((act) => act.value !== 0)
    .sort((a: IActivity, b: IActivity) => b.value - a.value),
  totalTime: 0
};

const cardData: ICardActivity = {
  activities: activities
    .filter((act) => act.value !== 0)
    .sort((a: IActivity, b: IActivity) => b.value - a.value),
  totalTime: totalTime
};

expect.extend({ toMatchImageSnapshot });

describe('Card  Component', () => {
  describe('Matches snapshots', () => {
    it('Matches not zero snapshot', () => {
      const card = renderer.create(<Card cardData={cardData} />).toJSON();
      expect(card).toMatchSnapshot();
    });
    it('Matches zero snapshot', () => {
      const card = renderer.create(<Card cardData={zeroCardData} />).toJSON();
      expect(card).toMatchSnapshot();
    });
  });
  describe('Matches total label and value', () => {
    it('Matches total value', () => {
      render(<Card cardData={cardData} />);
      expect(screen.getByTestId('totalTime')).toHaveTextContent(
        String(totalTime)
      );
    });
    it('Matches total label', () => {
      render(<Card cardData={cardData} />);
      expect(screen.getByTestId('totalTimeLabel')).toHaveTextContent(
        'Total time:'
      );
    });
  });
  describe('Matches activity labels and values', () => {
    it('Matches activity labels', () => {
      const { container } = render(<Card cardData={cardData} />);
      expect(container.getElementsByClassName('card-label')).toHaveLength(
        activities.length
      );
    });
    it('Matches activity values', () => {
      const { container } = render(<Card cardData={cardData} />);
      expect(container.getElementsByClassName('card-value')).toHaveLength(
        activities.length
      );
    });
  });
  describe('Matches zero activity labels and values', () => {
    it('Matches zero activity labels', () => {
      const { container } = render(<Card cardData={zeroCardData} />);
      expect(container.getElementsByClassName('card-label')).toHaveLength(1);
    });
    it('Matches zero activity values', () => {
      const { container } = render(<Card cardData={zeroCardData} />);
      expect(container.getElementsByClassName('card-value')).toHaveLength(1);
    });
  });
  describe('Matches sorted activities', () => {
    it('Matches max activity', () => {
      const { container } = render(<Card cardData={cardData} />);
      expect(container.getElementsByClassName('card-value')).toHaveLength(4);
      expect(
        container.getElementsByClassName('card-value')[1]
      ).toHaveTextContent(String(activities[0].value) + 'h');
    });
    it('Matches min actievity', () => {
      const { container } = render(<Card cardData={cardData} />);
      expect(container.getElementsByClassName('card-value')).toHaveLength(4);
      expect(
        container.getElementsByClassName('card-value')[3]
      ).toHaveTextContent(String(activities[2].value) + 'h');
    });
  });
});
describe('Pie React-Redux Component', () => {
  it('Matches canvas snapshot', async () => {
    const browser = await puppeteer.launch();
    const page = await browser.newPage();
    await page.waitFor(500);
    await page.goto('http://localhost:3000');
    await page.setViewport({
      width: 1366,
      height: 768,
      deviceScaleFactor: 1
    });
    const element = await page.$('.pie-chart');
    const image = await element?.screenshot();
    expect(image).toMatchImageSnapshot({ allowSizeMismatch: true });
    await browser.close();
  }, 30000);
});
