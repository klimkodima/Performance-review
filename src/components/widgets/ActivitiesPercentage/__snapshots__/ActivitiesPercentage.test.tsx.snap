// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card  Component Matches snapshots Matches not zero snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card-row"
  >
    <label
      className="card-label"
      data-testid="totalTimeLabel"
    >
      Total time
      :
    </label>
    <span
      className="card-value"
      data-testid="totalTime"
    >
      180
      h
    </span>
  </div>
  <div
    className="card-row"
  >
    <label
      className="card-label"
    >
      Audits
      :
    </label>
    <span
      className="card-value"
    >
      99
      h
    </span>
  </div>
  <div
    className="card-row"
  >
    <label
      className="card-label"
    >
      Meetings
      :
    </label>
    <span
      className="card-value"
    >
      63
      h
    </span>
  </div>
  <div
    className="card-row"
  >
    <label
      className="card-label"
    >
      Support
      :
    </label>
    <span
      className="card-value"
    >
      18
      h
    </span>
  </div>
</div>
`;

exports[`Card  Component Matches snapshots Matches zero snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card-row"
  >
    <label
      className="card-label"
      data-testid="totalTimeLabel"
    >
      Total time
      :
    </label>
    <span
      className="card-value"
      data-testid="totalTime"
    >
      0
      h
    </span>
  </div>
</div>
`;
